array_analyzer/extract/txt_parser.py

```diff
@@ -38,34 +53,124 @@ def create_xml_dict(path_):
+    Then, parses the .csvs and creates the four dictionaries:
+    :param path_: list
+        of strings to the .csv paths
+    :return:
+    """
+    # assign names to each of the types == params, spot types, antigens
+    fiduc = list()
+    csv_antigens = list()
+    array_params = dict()
+
+    for meta_csv in path_:
+        with open(meta_csv, newline='') as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=',')
+            for row in csv_reader:
+                # header row
+                if "Parameter" in str(row[0]) or '' in str(row[0]):
+                    continue
+
+                # parse params
+                elif "array_format_parameters" in meta_csv:
+                    array_params[row[0]] = row[1]
+
+                # parse fiducials
+                elif "array_format_type" in meta_csv:
+                    for col, value in enumerate(row[1:]):
+                        pos = {'@row': str(row[0]),
+                               '@col': str(col - 1),
+                               '@spot_type': str(value)}
+                        fiduc.append(pos)
+
+                # parse antigens
+                elif "array_format_antigen" in meta_csv:
+                    for col, value in enumerate(row[1:]):
+                        pos = {'@row': str(row[0]),
+                               '@col': str(col - 1),
+                               '@antigen': str(value)}
+                        csv_antigens.append(pos)
+
+    return fiduc, None, csv_antigens, array_params
+
+
+def create_xlsx_dict(path_):
+    """
+    extracts fiducial, antigen, and array parameter metadata from .xlsx sheets
+    then populates dictionaries or lists with appropriate information
+    :param path_: path to .xlsx file
+    :return list fiduc: Fiducials and control info
+    :return list spots: None.  spots IDs not needed for .xlsx
+    :return list repl: Replicate (antigen)
+    :return dict params: Additional parameters about hardware and array
+    """
+    fiduc = list()
+    xlsx_antigens = list()
+    array_params = dict()
+
+    xlsx = pd.read_excel(path_, sheet_name=None)
+
+    # populate array parameters
+    for idx, value in enumerate(xlsx['imaging_and_array_parameters']['Parameter']):
+        array_params[value] = xlsx['imaging_and_array_parameters']['Value'][idx]
+
+    # find and populate fiduc list, antigen list
+    for col in xlsx['array_antigens'].keys()[1:]:
+        for row, value in enumerate(xlsx['array_antigens'][col]):
+            if type(value) is float:
+                if math.isnan(value):
+                    continue
+            elif "Fiducial" in value:
+                pos = {'@row': row,
+                       '@col': col,
+                       '@spot_type': str(value)}
+                fiduc.append(pos)
+            else:
+                pos = {'@row': row,
+                       '@col': col,
+                       '@antigen': str(value)}
+                xlsx_antigens.append(pos)
+
```

Comment:
Same comment here as for csv. I'm not sure the point to convert the 2D pandas table to a dictionary... maybe I'm missing something here